src/Host/Broker/Impl/ApiErrorResult.cs

```diff
@@ -1,20 +1,24 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License. See LICENSE in the project root for license information.
 
+using System;
 using Microsoft.AspNetCore.Http;
 using Microsoft.AspNetCore.Mvc;
 using Microsoft.R.Host.Protocol;
 
 namespace Microsoft.R.Host.Broker {
     internal sealed class ApiErrorResult : ObjectResult {
-        public ApiErrorResult(HttpResponse response, BrokerApiError error) : base(error) {
+        public ApiErrorResult(HttpResponse response, BrokerApiError error, Exception ex = null) : base(error) {
```

Comment:
While you're here, can you also fix `ApiErrorResult` to not take `HttpResponse` as parameter? It's supposed to get one in its `ExecuteAsync` method, and modify it there, not in constructor. You can override it, call `base`, and then add the header.